### DISCUSSION TITLE
Поиск в поле селект

### DISCUSSION THREAD

**Discussion by mihail-cornev** [totumonline/totum-ru-issues-and-discussions#228] [Как это сделать?] [answered]
Привет,

Подскажите пожалуйста, если у меня есть поле с типом селект, с выборкой по клиентам из другой таблицы, но отображение в  этом поле только ФИО.
Суть в том, чтобы сотрудник случайно не создал второй раз одного и того же клиента, если будет искать только по имени к примеру.

Вопрос: Как сделать, чтобы в поле селект происходил поиск и по другим полям, типа email или телефон? Т.е. вместо имени сотрудник начнёт вводить номер телефона с которого ему звонят и сможет сразу проверить, если такой контакт создан.
Или может есть какое-то другое решение?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Можно скомбинировать отображение селекта через скрытое поле: https://ru.docs.totum.online/code-for-forming-a-select

**Reply by mihail-cornev** [reply]
Посмотрел видео, почитал про селекты ещё раз, но пока не разобрался.

**Reply by ttm-alexeypolunin** [reply]
Нужно, что бы в отображении Селекта было не только имя, но и телефон и email по которому будет осуществляться поиск:

Алексей [+1 999 555 00 00] totum@totum.online

Для этого в таблице источнике мы делаем техническое скрытое поле, в котором собираем нужные данные и именно это техническое поле берем в селект.

Вот видео с привязкой ко времени: https://youtu.be/qmVTZ19rW1o?t=304

**Reply by mihail-cornev** [reply]
Да, это я понял. Но можно ли сделать, чтобы в селекте только ФИО отображались, без email и телефон, и при этом поиск можно было бы делать по телефону и email. А то очень длинно получается "имя - телефон - email".
Либо чтобы они отображались по раскрытию селекта.

**Reply by ttm-alexeypolunin** [reply]
В коде форматирования:

```
f1=: setFormat(condition: $#nfv != ""; text: $select)

select: select(table: 'table_for_select'; field: 'fio'; where: 'id' = $#nfv)
```

Но тут надо понимать, что это `+ 1` запрос на каждую отображаемую строку с этим Селектом. Те если это одно поле в хедере например внутри цикла, то ок. Если это столбец в таблице Циклов, то там так лучше не делать, а сделать селект, который берет только `fio`.

**Reply by mihail-cornev** [reply]
Спасибо за помощь, интересное решение. Понял, суть в том, чтобы сделать просто насколько возможно чтобы не нагружать систему.

**Reply by ttm-alexeypolunin** [reply]
В таком варианте даже если на 1К строк будет таблица то это будет дополнительные 300-400 мс (что не круто если с таблицей постоянно работают). Но если там будет 50К строк в выводе на странице, вот тут-то все ляжет.
